src/main/scala/org/broadinstitute/dsde/firecloud/service/StorageService.scala

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import akka.actor._
+import akka.pattern._
+import akka.event.Logging
+import org.broadinstitute.dsde.firecloud.{FireCloudExceptionWithErrorReport, Application}
+import org.broadinstitute.dsde.firecloud.dataaccess._
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.service.PerRequest._
+import org.broadinstitute.dsde.firecloud.utils.RestJsonClient
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import scala.concurrent.ExecutionContext
+
+/**
+ * Created by mbemis on 11/28/16.
+ */
+object StorageService {
+  sealed trait StorageServiceMessage
+  case class GetObjectStats(bucketName: String, objectName: String) extends StorageServiceMessage
+
+  def props(storageServiceConstructor: UserInfo => StorageService, userInfo: UserInfo): Props = {
+    Props(storageServiceConstructor(userInfo))
+  }
+
+  def constructor(app: Application)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
+    new StorageService(userInfo, app.googleServicesDAO)
+}
+
+class StorageService(protected val argUserInfo: UserInfo, val googleServicesDAO: GoogleServicesDAO)(implicit val executionContext: ExecutionContext) extends Actor with RestJsonClient {
+  implicit val system = context.system
+  val log = Logging(system, getClass)
+  implicit val userInfo = argUserInfo
+  import StorageService._
+
+  override def receive: Receive = {
+    case GetObjectStats(bucketName: String, objectName: String) => getObjectStats(bucketName, objectName) pipeTo sender
+  }
+
+  def getObjectStats(bucketName: String, objectName: String) = {
+    requestToObject[JsObject](googleServicesDAO.getObjectMetadataRequest(bucketName, objectName)) flatMap { googleResponse =>
+      googleServicesDAO.fetchPriceList map { googlePrices =>
+        googleResponse.getFields("size").headOption match {
+          case Some(size) => {
+            val fileSizeGB = BigDecimal(size.toString.replaceAll("\"", "")) / Math.pow(1000, 3)
+            val egressPrice = getEgressCost(googlePrices.prices.cpComputeengineInternetEgressNA.tiers, fileSizeGB, 0)
+            RequestComplete(StatusCodes.OK, googleResponse.copy(googleResponse.fields ++ Map("estimatedCostUSD" -> JsNumber(egressPrice))))
+          }
+          case None => RequestComplete(StatusCodes.OK, googleResponse)
+
+        }
+      }
+    }
+  }
+
+  private def getEgressCost(googlePrices: Map[Long, BigDecimal], fileSizeGB: BigDecimal, totalCost: BigDecimal): BigDecimal = {
```

Comment:
Is there some documentation that describes this cost calculation? What it looks like you've implemented is something like income tax brackets where the first x GB are charged at one rate, the next y GB at another rate, etc. To me, the pricing scheme at https://cloud.google.com/storage/pricing implies that all egress is charged at a single rate that is determined by monthly usage.